modules/etcd-creating-automated-backups.adoc

```diff
@@ -0,0 +1,306 @@
+spec:
+  capacity:
+    storage: 100Gi <.>
+  volumeMode: Filesystem
+  accessModes:
+  - ReadWriteMany
+  persistentVolumeReclaimPolicy: Delete
+  storageClassName: local-storage
+  local:
+    path: /mnt/
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - <example-master-node> <.>
+----
+<.> The amount of storage available to the PV. Adjust this value for your requirements.
+<.> Replace this value with the node to attach this PV to.
++
+[TIP]
+====
+Run the following command to list the available nodes:
+
+[source,terminal]
+----
+$ oc get nodes
+----
+====
+
+.. Verify the creation of the PV by running the following command:
++
+[source,terminal]
+----
+$ oc get pv
+----
++
+.Example output
+[source,terminal]
+----
+NAME                    CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS    REASON   AGE
+etcd-backup-pv-fs       100Gi      RWX            Delete           Available           local-storage            10s
+----
+
+.. Create a PVC named `etcd-backup-pvc.yaml` with contents such as the following:
++
+[source,yaml]
+----
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: etcd-backup-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 10Gi <.>
+  storageClassName: local-storage
+----
+<.> The amount of storage available to the PVC. Adjust this value for your requirements.
+
+.. Apply the PVC by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f etcd-backup-pvc.yaml
+----
+
+. Create a custom resource definition (CRD) file named `etcd-recurring-backups.yaml`. The contents of the created CRD define the schedule and retention type of automated backups.
++
+For the default retention type, use contents such as the following:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: etcd-recurring-backup
+spec:
+  etcd:
+    schedule: "20 4 * * *" <.>
+    timeZone: "UTC"
+    retentionPolicy:
+      retentionType: "" <.>
+    pvcName: etcd-backup-pvc
+----
+<.> The `CronTab` schedule for recurring backups. Adjust this value for your needs.
+<.> The `""` retention type specifies the default type and size.
++
+To use retention based on the maximum number of backups, add the following key-value pairs to the `retentionPolicy`:
++
+[source,yaml]
+----
+retentionPolicy:
+  retentionType: RetentionNumber
+  retentionNumber:
+    maxNumberOfBackups: 5 <.>
+----
+<.> The maximum number of backups to retain. Adjust this value for your needs.
++
+For retention based on the file size of backups, use the following:
++
+[source,yaml]
+----
+retentionPolicy:
+  retentionType: RetentionSize
+  retentionSize:
+    maxSizeOfBackupsGb: 10 <.>
+----
+<.> The maximum file size of the retained backups in gigabytes. Adjust this value for your needs.
```

Comment:
please leave a note around a known bug for the retention. Both size and number can go over the configured limit - that's known. Just in case people configure 10gig PVCs and they run out of space immediately.

One snapshot can be up to 8 gigs, so there needs to be enough buffer left.

seems this memo got lost between tickets :/ 

https://issues.redhat.com/browse/OCPBUGS-17952?focusedId=23173156&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-23173156

Reply:
The issue linked to only discusses `maxNumberOfBackups` and explains that the reality is one backup higher than configured. Do we have something specific to say about `maxSizeOfBackupsGb`? Saying that it "can be higher than configured" seems vague to me. If we have nothing specific to say, I'm okay with it.

Reply:
> Do we have something specific to say about maxSizeOfBackupsGb? 

I'd tend to say  +10 gigs over the configured retention would be safe.